src/test/util_tests.cpp

```diff
@@ -200,23 +203,15 @@ BOOST_AUTO_TEST_CASE(parse_hex)
 
 BOOST_AUTO_TEST_CASE(util_HexStr)
 {
-    BOOST_CHECK_EQUAL(
-        HexStr(ParseHex_expected),
-        "04678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5f");
-
-    BOOST_CHECK_EQUAL(
-        HexStr(Span{ParseHex_expected}.last(0)),
-        "");
-
-    BOOST_CHECK_EQUAL(
-        HexStr(Span{ParseHex_expected}.first(0)),
-        "");
+    BOOST_CHECK_EQUAL(HexStr(HEX_PARSE_OUTPUT), HEX_PARSE_INPUT);
+    BOOST_CHECK_EQUAL(HexStr(Span{HEX_PARSE_OUTPUT}.last(0)), "");
+    BOOST_CHECK_EQUAL(HexStr(Span{HEX_PARSE_OUTPUT}.first(0)), "");
 
     {
-        const std::vector<char> in_s{ParseHex_expected, ParseHex_expected + 5};
+        constexpr char out_exp[] = "04678afdb0";
```

Comment:
My goal was to make it `constexpr`. (`MakeUCharSpan` and `MakeByteSpan` used on the lines below are faux-`constexpr` since they call functions that cannot be `constexpr` as they use `reinterpret_cast`).

A possible but noisier alternative, keeping `string` would be:
```C++
static constexpr std::string out_exp{"04678afdb0"};
```
`string_view` might be a good compromise:
```C++
constexpr std::string_view out_exp{"04678afdb0"};
```
I general having a plain C-array is more terse and narrows what the programmer (and compiler) can expect the type to do, so I prefer it in cases like this. But will switch to `string_view` if I retouch.